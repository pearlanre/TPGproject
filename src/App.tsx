import React, { useEffect, useState } from 'react';
import Select from 'react-select';
import './App.css';
import { ILoyalty } from './interfaces/ILoyalty';
import { AirlinePointsOrCash } from './components/AirlinePointsOrCash/AirlinePointsOrCash';
import { HotelsPointsOrCash } from './components/HotelsPointsOrCash/HotelsPointsOrCash';

function App(): any {
  const [airlineData, setAirlineData] = useState<ILoyalty>({
    name: '',
    type: '',
    tpg_valuation: 0,
  });
  const [hotelData, setHotelData] = useState<ILoyalty>({
    name: '',
    type: '',
    tpg_valuation: 0,
  });
  const [selectedOption, setSelectedOption] = useState(null);
  const [showAccordion, setShowAccordion] = useState(false);
  const getData = () => {
    fetch('data.json', {
      headers: {
        'Content-Type': 'application/json',
        Accept: 'application/json',
      },
    })
      .then(function (response) {
        return response.json();
      })
      .then(function (myJson) {
        const airlineFilter = myJson.filter(
          (val) => val.tpg_valuation !== '' && val.type === 'airline'
        );
        const hotelFilter = myJson.filter(
          (val) => val.tpg_valuation !== '' && val.type === 'hotel'
        );
        setAirlineData(airlineFilter);
        setHotelData(hotelFilter);
      });
  };
  const options = [
    { id: 0, name: 'Air Tickets' },
    { id: 1, name: 'Hotels' },
  ];
  const handleSelection = (option) => {
    setSelectedOption(option.id);
    setShowAccordion(true);
  };
  useEffect(() => {
    getData();
  }, []);
  // @ts-ignore
  return (
    <div className="App">
      <h1 className="tpg_title">Cash to Points Calculator</h1>
      <p>Select Air Tickets or Hotels</p>
      <Select
        defaultValue={selectedOption}
        onChange={handleSelection}
        options={options}
        getOptionLabel={(option) => option.name}
        getOptionValue={(option) => option.id}
      />
      {showAccordion &&
        (selectedOption === 1 ? (
          <HotelsPointsOrCash data={hotelData} />
        ) : (
          <AirlinePointsOrCash data={airlineData} />
        ))}
    </div>
  );
}

export default App;
