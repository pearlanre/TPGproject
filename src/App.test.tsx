import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('Point to Cash App component', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<App />);
  });
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render the PointsOrCash Component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('renders App component', () => {
    expect(wrapper.find('.App')).toHaveLength(1);
  });
  it('renders heading', () => {
    wrapper = shallow(<App />);
    const heading = 'Cash to Points Calculator';
    expect(wrapper.find('h1').text()).toEqual(heading);
  });
  it('renders Select Text', () => {
    wrapper = shallow(<App />);
    const heading = 'Select Air Tickets or Hotels';
    expect(wrapper.find('p').text()).toEqual(heading);
  });
});
