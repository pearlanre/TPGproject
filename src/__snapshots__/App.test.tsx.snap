// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Point to Cash App component should render the PointsOrCash Component correctly 1`] = `
<div
  className="App"
>
  <h1
    className="tpg_title"
  >
    Cash to Points Calculator
  </h1>
  <p>
    Select Air Tickets or Hotels
  </p>
  <StateManager
    defaultInputValue=""
    defaultMenuIsOpen={false}
    defaultValue={null}
    getOptionLabel={[Function]}
    getOptionValue={[Function]}
    onChange={[Function]}
    options={
      Array [
        Object {
          "id": 0,
          "name": "Air Tickets",
        },
        Object {
          "id": 1,
          "name": "Hotels",
        },
      ]
    }
  />
</div>
`;
